tests/integration/venv_ITs/test_issue_2160.py

```diff
@@ -0,0 +1,126 @@
+from pex.testing import WheelBuilder, run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, List
+
+
+@pytest.fixture
+def top_level_wheel(tmpdir):
+    # type: (Any) -> str
+    top_level_project = os.path.join(str(tmpdir), "project")
+
+    top_level_library = os.path.join(top_level_project, "top_level", "lib.py")
+    with safe_open(top_level_library, "w") as fp:
+        fp.write("OG = 'Henry Barber'")
+
+    with safe_open(os.path.join(top_level_project, "pyproject.toml"), "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                [build-system]
+                requires = ["setuptools"]
+                build-backend = "setuptools.build_meta"
+
+                [project]
+                name = "top_level"
+                version = "0.1.0"
+                """
+            )
+        )
+
+    wheel_dir = os.path.join(str(tmpdir), "wheels")
+    return WheelBuilder(source_dir=top_level_project, wheel_dir=wheel_dir).bdist()
+
+
+@pytest.mark.parametrize(
+    "style_args",
+    [
+        pytest.param([], id="zipapp"),
+        pytest.param(["--venv"], id="venv (site-packages symlinks)"),
+        pytest.param(["--venv", "--venv-site-packages-copies"], id="venv (site-packages copies)"),
+    ],
+)
+@pytest.mark.parametrize(
+    "layout", [pytest.param(layout, id=layout.value) for layout in Layout.values()]
+)
+def test_ns_package_split_across_sources_and_deps(
+    tmpdir,  # type: Any
+    top_level_wheel,  # type: str
+    style_args,  # type: List[str]
+    layout,  # type: Layout.Value
+):
+    # type: (...) -> None
+
+    sources = os.path.join(str(tmpdir), "sources")
+    with safe_open(os.path.join(sources, "top_level", "mymain.py"), "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                from top_level.lib import OG
+
+
+                print("Hello {}!".format(OG))
+                """
+            )
+        )
+
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+    pex = os.path.join(str(tmpdir), "binary.pex")
+    run_pex_command(
+        args=[
+            "--pex-root",
+            pex_root,
+            "--runtime-pex-root",
+            pex_root,
+            top_level_wheel,
+            "-D",
+            sources,
+            "-e",
+            "top_level.mymain",
+            "-o",
+            pex,
+            "--layout",
+            layout.value,
+        ]
+        + style_args
+    ).assert_success()
+
+    safe_rmtree(pex_root)
+    assert (
+        "Hello Henry Barber!"
+        == subprocess.check_output(args=[sys.executable, pex]).decode("utf-8").strip()
+    )
+
+    distributions = list(PEX(pex).resolve())
+    assert 1 == len(distributions)
+
+    top_level = distributions[0]
+    assert ProjectName("top_level") == top_level.metadata.project_name
+    assert Version("0.1.0") == top_level.metadata.version
+
+    assert [os.path.join(top_level.location, "top_level", "lib.py")] == [
```

Comment:
This test fails pre-fix for 2/9 cases as expected: symlinked venv {packed,zipapp}